tests/functional/filter/cram/subsample-weighted-and-uniform-mix.t

```diff
@@ -0,0 +1,109 @@
+Setup
+
+  $ source "$TESTDIR"/_setup.sh
+
+Generate metadata file with 250 rows.
+
+  $ echo "strain	date	location" > metadata.tsv
+  $ for i in $(seq 1 50); do
+  >     echo "2000A_$i	2000	A" >> metadata.tsv
+  >     echo "2000B_$i	2000	B" >> metadata.tsv
+  >     echo "2001A_$i	2001	A" >> metadata.tsv
+  >     echo "2001B_$i	2001	B" >> metadata.tsv
+  >     echo "2002B_$i	2002	B" >> metadata.tsv
+  > done
+
+Weight locations A:B as 2:1. This is reflected in target_group_sizes.tsv below.
+
+  $ cat >weights.tsv <<~~
+  > location	weight
+  > A	2
+  > B	1
+  > ~~
+
+  $ ${AUGUR} filter \
+  >   --metadata metadata.tsv \
+  >   --group-by location \
+  >   --group-by-weights weights.tsv \
+  >   --subsample-max-sequences 100 \
+  >   --subsample-seed 0 \
+  >   --output-group-by-sizes target_group_sizes.tsv \
+  >   --output-strains strains.txt 2>/dev/null
+
+  $ cat target_group_sizes.tsv
+  location	weight	_augur_filter_target_size_int
+  A	2	67
+  B	1	33
```

Comment:
Any reason not to also check the actual number of strains? e.g. `grep 'A_' < strains.txt | wc -l`. It should be stable for a given `--subsample-seed` I think?

Reply:
How about checking the grouped column values in output metadata instead of strains? Added that in 4f7738daa95c9e68087ffe2bbe81ee5556dbc5c3

Reply:
Perfect!